lib/router.js

```diff
@@ -120,7 +120,12 @@ Route.prototype.match = function (url, options) {
     // 4. method/path/navParams all matched, extract the matched path params
     var routeParams = {};
     for (i = 0, len = self.keys.length; i < len; i++) {
-        routeParams[self.keys[i].name] = pathMatches[i+1];
+        // Don't overwrite a previously populated parameter with `undefined`.
+        // A route may legitimately have multiple instances of a parameter
+        // name if the path was an array.
+        if (pathMatches[i+1] !== void 0 || routeParams[self.keys[i].name] === void 0){
```

Comment:
minor:  instead of `void 0`, could you use `undefined` to be consistent with rest of code?